npc/re/jobs/novice/academy.txt

```diff
@@ -12969,33 +12969,17 @@ izlude_d,153,126,1	duplicate(Refinery Owner Han#iz)	Refinery Owner Han#iz_d	4_M_
 			}
 		}
 	}
-	switch(getequipweaponlv(.@part)) {
-	case 0:
+
+	.@refineitemid = getequipid(.@part); // save id of the item
+	.@refinerycnt = getequiprefinerycnt(.@part); //save refinery count
+	.@price = getequiprefinecost(.@part, REFINE_COST_WAGJAK, REFINE_ZENY_COST);
+	.@material = getequiprefinecost(.@part, REFINE_COST_WAGJAK, REFINE_MATERIAL_ID);
+
+	if ((getequipweaponlv(.@part) >= 1) && (getequipweaponlv(.@part) <= 4))
+		.@type$ = "Weapon";
```

Comment:
In the previous version type display the weapon level
`.@type$ = "Level " + getequipweaponlv(.@part) + " weapon";`